### PR TITLE
Update sample to follow Basher and Zero Trust

### PR DESCRIPTION
**Pull request by salman90**
# Pull Request

## Purpose

Update sample to follow Basher and Zero Trust

## Does this introduce a breaking change

```console
    [ x ] Yes
    [ ] No
```

## Pull request type

What kind of change does this Pull Request introduce?

```console
    [ ] Bugfix
    [ x ] Feature
    [ x ] Code style update (formatting, local variables)
    [ x ] Documentation content changes
    [ ] Other... Please describe:
```

## How to test

* Get the code

```console
    git clone [repo-address]
    cd [repo-name]
    git checkout [branch-name]
    npm install
```

* Test the code

```console

```

## What to check

ex: verify that the following are valid:

* ...

### CODE REVIEW

File: 1-Authentication/2-sign-in-b2c/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    "AADApps": [
+        {
+            "Id": "client",
+            "Name": "ms-identity-javascript-c1s2",
+            "Kind": "SinglePageApplication",
+            "Audience": "AzureADandPersonalMicrosoftAccount",
+            "HomePage": "http://localhost:6420",
+            "SampleSubPath": "1-Authentication\\2-sign-in-b2c",
+            "ReplyUrls": "http://localhost:6420, http://localhost:6420/redirect.html"
```
Comment (derisen) [inline 1-Authentication/2-sign-in-b2c/AppCreationScripts/sample.json:23]:
```suggestion
            "ReplyUrls": "http://localhost:6420, http://localhost:6420/redirect"
```

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 2]
```diff
@@ -54,70 +71,78 @@ From your shell or command line:
...
 1. Find the key `authorities` and populate it with your policy authority strings e.g. `https://<your-tenant-name>.b2clogin.com/<your-tenant-name>.onmicrosoft.com/b2c_1_susi`.
 1. Find the key `authorityDomain` and populate it with the domain portion of your authority string e.g. `<your-tenant-name>.b2clogin.com`.
 
-## Running the sample
+
+### Step 4: Running the sample
 
 ```console
-    cd 1-Authentication/2-sign-in-b2c
+    cd 1-Authentication\2-sign-in-b2c
```
Comment (kalyankrishna1) [inline 1-Authentication/2-sign-in-b2c/README.md:145]: >1-Authentication\2-sign-in-b2c [](http://example.com/codeflow?start=7&length=30)

sample.json as a SampleSubPath defined, but this was not generated, please investigate

File: 1-Authentication/2-sign-in-b2c/server.js [hunk 3]
```diff
@@ -2,11 +2,31 @@ const express = require('express');
...
 const app = express();
 
+/**
+ * HTTP request handlers should not perform expensive operations such as accessing the file system,
+ * executing an operating system command or interacting with a database without limiting the rate at
+ * which requests are accepted. Otherwise, the application becomes vulnerable to denial-of-service attacks
+ * where an attacker can cause the application to crash or become unresponsive by issuing a large number of
+ * requests at the same time. For more information, visit: https://cheatsheetseries.owasp.org/cheatsheets/Denial_of_Service_Cheat_Sheet.html
+ */
+const limiter = rateLimit({
```
Comment (derisen) [inline 1-Authentication/2-sign-in-b2c/server.js:19]: This is not a web API sample, and the server here is for demonstration only -we shouldn't bother with rate-limiting

File: 1-Authentication/2-sign-in-b2c/App/authPopup.js [hunk 4]
```diff
@@ -2,13 +2,74 @@
...
+         * policies may use "acr" instead of "tfp"). To learn more about B2C tokens, visit:
+         * https://docs.microsoft.com/en-us/azure/active-directory-b2c/tokens-overview
+         */
+
+        if (event.payload.idTokenClaims['tfp'] === b2cPolicies.names.editProfile) {
+            const originalSignInAccount = myMSALObj
+                .getAllAccounts()
+                .find(
+                    (account) =>
+                        account.idTokenClaims.oid === event.payload.idTokenClaims.oid &&
```
Comment (derisen) [inline 1-Authentication/2-sign-in-b2c/App/authPopup.js:27]: what happens if/when oid claim is not available?

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 5]
```diff
@@ -259,81 +307,84 @@ In case if you are using redirect flow, you should catch the error inside `handl
...
+                })
     }
 ```
 
-- **Edit Profile**
+* **Edit Profile**
 
-Unlike password reset, edit profile user-flow does not require users to sign-out and sign-in again. Instead, **MSAL.js** will handle
-switching back to the authority string of the default user-flow automatically.
+hen a user selects the **Edit Profile** button on the navigation bar, we simply initiate a sign-in flow. Like password reset, edit profile user-flow requires users to sign-out and sign-in again.
```
Comment (derisen) [inline 1-Authentication/2-sign-in-b2c/README.md:327]:
```suggestion
When a user selects the **Edit Profile** button on the navigation bar, we simply initiate a sign-in flow. Like password reset, edit profile user-flow requires users to sign-out and sign-in again.
```

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 6]
```diff
@@ -259,81 +307,84 @@ In case if you are using redirect flow, you should catch the error inside `handl
...
-If you find a bug in the sample, please raise the issue on [GitHub Issues](../../issues).
-
-To provide a recommendation, visit the following [User Voice page](https://feedback.azure.com/forums/169401-azure-active-directory).
+* [Vanilla JavaScript single-page application (SPA) using MSAL.js to authorize users for calling a protected web API on Azure AD B2C](https://github.com/Azure-Samples/ms-identity-javascript-tutorial/tree/main/3-Authorization-II/2-call-api-b2c)
+* [JavaScript single-page application calling Microsoft Graph with delegated permissions to manage Azure AD B2C user accounts](https://github.com/Azure-Samples/ms-identity-javascript-tutorial/tree/main/2-Authorization-I/2-call-graph-b2c)
 
 ## Contributing
 
-If you'd like to contribute to this sample, see [CONTRIBUTING.MD](../../CONTRIBUTING.md).
+If you'd like to contribute to this sample, see [CONTRIBUTING.MD](/CONTRIBUTING.md).
```
Comment (derisen) [inline 1-Authentication/2-sign-in-b2c/README.md:373]: > If you'd like to contribute to this sample, see [CONTRIBUTING.MD](/CONTRIBUTING.md)

is the file location correct here?

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 7]
```diff
@@ -54,70 +71,78 @@ From your shell or command line:
...
 1. Find the key `authorityDomain` and populate it with the domain portion of your authority string e.g. `<your-tenant-name>.b2clogin.com`.
 
-## Running the sample
+
+### Step 4: Running the sample
 
 ```console
-    cd 1-Authentication/2-sign-in-b2c
+    cd 1-Authentication\2-sign-in-b2c
     npm start
```
Comment (kalyankrishna1) [inline 1-Authentication/2-sign-in-b2c/README.md:146]: >npm start [](http://example.com/codeflow?start=4&length=9)

it never starts unless i do the
npm install
first :unamused:

**Comment by kalyankrishna1** [comment]
sign-in with gitHub creates a "Authorize derisen" popup..

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: 1-Authentication/2-sign-in-b2c/App/authPopup.js [hunk 8]
```diff
@@ -2,13 +2,74 @@
...
+                        account.idTokenClaims['tfp'] === b2cPolicies.names.signUpSignIn
+                );
+
+            let signUpSignInFlowRequest = {
+                authority: b2cPolicies.authorities.signUpSignIn.authority,
+                account: originalSignInAccount,
+            };
+
+            // silently login again with the signUpSignIn policy
+            myMSALObj.ssoSilent(signUpSignInFlowRequest).catch((error) => {
```
[the remark this answers is not in the document]
Comment (salman90) [reply 1-Authentication/2-sign-in-b2c/App/authPopup.js]: Done!

File: 1-Authentication/2-sign-in-b2c/README.md (hunk 2 above)
Comment (salman90) [reply 1-Authentication/2-sign-in-b2c/README.md:145]: It works fine when I run the codegen

File: 1-Authentication/2-sign-in-b2c/README.md (hunk 7 above)
Comment (salman90) [reply 1-Authentication/2-sign-in-b2c/README.md:146]: The [second step ](cd 1-Authentication\2-sign-in-b2c)in the setup is to run npm insall

File: 1-Authentication/2-sign-in-b2c/App/authPopup.js (hunk 4 above)
Comment (salman90) [reply 1-Authentication/2-sign-in-b2c/App/authPopup.js]: Updated!.

**Review by derisen — APPROVED** [review]

**Pull request merged**